File: app/src/main/java/com/websarva/wings/android/zuboradiary/ui/list/diarylist/DiaryDayList.java
package com.websarva.wings.android.zuboradiary.ui.list.diarylist;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Objects;

public class DiaryDayList {

    private final List<DiaryDayListItem> diaryDayListItemList;

    public DiaryDayList(List<DiaryDayListItem> itemList) {
        Objects.requireNonNull(itemList);
        if (itemList.isEmpty()) throw new IllegalArgumentException();
        itemList.stream().forEach(Objects::requireNonNull);

        this.diaryDayListItemList = Collections.unmodifiableList(itemList);
    }

    public DiaryDayList() {
        this.diaryDayListItemList = new ArrayList<>();
    }

    public int countDiaries() {
        return diaryDayListItemList.size();
    }

    public DiaryDayList combineDiaryDayLists(DiaryDayList additionList) {
        Objects.requireNonNull(additionList);
        if (additionList.diaryDayListItemList.isEmpty()) throw new IllegalArgumentException();

        List<DiaryDayListItem> resultItemList = new ArrayList<>();
        resultItemList.addAll(this.diaryDayListItemList);
        resultItemList.addAll(additionList.diaryDayListItemList);

        return new DiaryDayList(resultItemList);
    }

    public List<DiaryDayListItem> getDiaryDayListItemList() {
        return diaryDayListItemList;
    }
}

File: app/src/main/java/com/websarva/wings/android/zuboradiary/ui/list/diarylist/DiaryDayList.kt
package com.websarva.wings.android.zuboradiary.ui.list.diarylist

class DiaryDayList {

    val diaryDayListItemList: List<DiaryDayListItem>

    constructor(itemList: List<DiaryDayListItem>) {
        require(itemList.isNotEmpty())

        this.diaryDayListItemList = itemList.toList()
    }

    constructor(): this(ArrayList())

    fun countDiaries(): Int {
        return diaryDayListItemList.size
    }

    fun combineDiaryDayLists(additionList: DiaryDayList): DiaryDayList {
        require(additionList.diaryDayListItemList.isNotEmpty())

        val resultItemList = diaryDayListItemList + additionList.diaryDayListItemList
        return DiaryDayList(resultItemList)
    }
}
